src/modules/job-manager/event.c

```diff
@@ -240,29 +243,46 @@ int event_job_update (struct job *job, const char *event)
         if (job->state != FLUX_JOB_NEW)
             goto inval;
         job->t_submit = timestamp;
-        if (util_int_from_context (context, "priority", &priority) < 0)
+        if (!(o = json_loads (context, 0, NULL))) {
```

Comment:
This function is going to get quite long as we add events.  Could we perhaps create functions to decode each event that has a context, e.g. for this one,  maybe
```c
int event_submit_context_decode (const char *event, int *priority, uint32_t *userid, int *flags);
```